### DISCUSSION TITLE
How to rotate logs in a container environment?

### DISCUSSION THREAD

**Discussion by patrickdung** [ClickHouse/ClickHouse#47497] [Q&A] [answered]
The Clickhouse is deployed in Kubernetes.
How to rotate the log and error log in ```/var/log/clickhouse-server/clickhouse-server.log``` ?

For non-container environment, it uses logrotate for it.
How about in a container environment?

**Comment by mlex** [comment, marked as answer]
You can use the ClickHouse builtin log rotation.

It can be configured in the `logger` section of the server log. You can have a look at the default configuration (which is also used in the `clickhouse-server` docker image) [in the code](https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/config.xml#L30-L31).

```
<clickhouse>
    <logger>
            <!-- Rotation policy
             See https://github.com/pocoproject/poco/blob/poco-1.9.4-release/Foundation/include/Poco/FileChannel.h#L54-L85
          -->
          <size>1000M</size>
          <count>10</count>
    </logger>
</clickhouse>
```


You can override this when you run the clickhouse-server docker container by mounting your own additonal config file. For example to only keep 5 (gzipped) old logfiles with each max 10MB, you can use:
```
$ cat /tmp/logrotate.xml
<clickhouse>
  <logger>
    <size>10M</size>
    <count>5</count>
  </logger>
</clickhouse>

$ docker run -d --name some-clickhouse-server -v /tmp/logrotate.xml:/etc/clickhouse-server/config.d/logrotate.xml clickhouse/clickhouse-server
```

**Comment by patrickdung** [comment]
@mlex Thanks Michael
